src/js/features/content-generation/fabian.js

```diff
@@ -0,0 +1,1130 @@
+			},
+		];
+		setConversation( updatedConversation );
+
+		// Call API
+		setIsLoading( true );
+		apiFetch( {
+			path: '/classifai/v1/create-content',
+			method: 'POST',
+			data: {
+				id: postId,
+				summary: prompt,
+				title,
+				conversation: updatedConversation.slice( 0, -1 ),
+			},
+		} ).then(
+			( res ) => {
+				// Update conversation with response
+				setConversation( [
+					...updatedConversation.slice( 0, -1 ),
+					{
+						prompt,
+						completion: res,
+					},
+				] );
+				setError( false );
+				setIsLoading( false );
+			},
+			( err ) => {
+				setError( err?.message || 'An error occurred' );
+				setIsLoading( false );
+			}
+		);
+	};
+
+	const handleSubmit = ( event ) => {
+		event.preventDefault();
+		if ( ! inputValue.trim() ) {
+			return;
+		}
+
+		const userMessage = inputValue;
+		setInputValue( '' );
+
+		// Get post data
+		const postId = select( editorStore ).getCurrentPostId();
+		const title = select( editorStore ).getEditedPostAttribute( 'title' );
+
+		// Update conversation immediately with user message
+		const updatedConversation = [
+			...conversation,
+			{
+				prompt: userMessage,
+				completion: null, // Will be filled in once API response is received
+			},
+		];
+		setConversation( updatedConversation );
+
+		// Call API
+		setIsLoading( true );
+		apiFetch( {
+			path: '/classifai/v1/create-content',
+			method: 'POST',
+			data: {
+				id: postId,
+				summary: userMessage,
+				title,
+				conversation: updatedConversation.slice( 0, -1 ), // Exclude the message we just added
+			},
+		} ).then(
+			( res ) => {
+				// Update conversation with response
+				setConversation( [
+					...updatedConversation.slice( 0, -1 ),
+					{
+						prompt: userMessage,
+						completion: res,
+					},
+				] );
+				setError( false );
+				setIsLoading( false );
+			},
+			( err ) => {
+				setError( err?.message || 'An error occurred' );
+				setIsLoading( false );
+			}
+		);
+	};
+
+	const handleKeyDown = ( event ) => {
+		// Submit on Enter key, but not when Shift is pressed
+		if ( event.key === 'Enter' && ! event.shiftKey ) {
+			event.preventDefault();
+			handleSubmit( event );
+		}
+		// Shift+Enter will add a new line by default (no action needed)
+	};
+
+	const toggleChatUI = () => {
+		setIsExpanded( ! isExpanded );
+	};
+
+	const startOver = () => {
+		setConversation( [] );
+		setError( false );
+	};
+
+	const insertContent = ( content ) => {
+		dispatch( editorStore )
+			.editPost( {
+				content: '',
+			} )
+			.then( () => {
+				const contentWithEntities = decodeEntities( content );
+
+				const containsBlockMarkup =
+					contentWithEntities.includes( '<!-- wp:' );
+
+				let blocks;
+				if ( containsBlockMarkup ) {
+					blocks = parse( contentWithEntities );
+				} else {
+					blocks = pasteHandler( {
+						HTML: contentWithEntities,
+						plainText: contentWithEntities,
+						mode: 'BLOCKS',
+					} );
```

Comment:
@dkotter Rather then the RawHandler with autop I found this to give me much better results when it comes to markdown parsing / handling other types of rich text.